### PR TITLE
Add support for providing a custom "fields" parameter to getResourceOwnerDetailsUrl()

### PR DESCRIPTION
**Pull request by zegenie**
This implements a fix for issue #71

### CODE REVIEW

File: src/Provider/Facebook.php [hunk 1]
```diff
@@ -104,17 +104,23 @@ public function getDefaultScopes(): array
         return ['public_profile', 'email'];
     }
 
-    public function getResourceOwnerDetailsUrl(AccessToken $token): string
+    public function getResourceOwnerDetailsUrl(AccessToken $token, array $fields = null): string
```
Comment (shadowhand) [inline src/Provider/Facebook.php:107]: This modifies the signature of [AbstractProvider::getResourceOwnerDetailsUrl][1]. Instead, I suggest that `fields` be added as a parameter to the constructor `$options`.

[1]: https://github.com/thephpleague/oauth2-client/blob/2.6.0/src/Provider/AbstractProvider.php

**Review by shadowhand — CHANGES_REQUESTED** [review]

File: src/Provider/Facebook.php [hunk 2]
```diff
@@ -87,6 +94,25 @@ public function __construct($options = [], array $collaborators = [])
         if (!empty($options['enableBetaTier']) && $options['enableBetaTier'] === true) {
             $this->enableBetaMode = true;
         }
+
+        if (!empty($options['fields']) && is_array($options['fields'])) {
+            array_walk($options['fields'], static function ($value) { if (!is_string($value)) {
+                throw new FacebookProviderException('Facebook does not support non-string fields');
+            }});
```
Comment (shadowhand) [inline src/Provider/Facebook.php:101]: I don't think this is necessary. Configuration should be the same dev/prod, and someone will realize pretty quickly if the fields are not valid.

Comment (zegenie) [reply src/Provider/Facebook.php]: Fair enough, I wasn't sure which level of validation was desired. PR updated.

**Comment by NelinD** [comment]
When will the release be?

**Comment by shadowhand** [comment]
@NelinD my apologies, I did not realize this was never tagged. Release 2.1.0 is now available.

**Comment by NelinD** [comment]
Thank you very much!

**Pull request merged**